### PR TITLE
Added a section about breaking changes that require approval

### PR DESCRIPTION
**Pull request by paliarush**
## This PR is a:

- Bug fix or improvement

## Summary

Defined a process of architectural approval for the backward-incompatible changes on code level

## Additional information

List all affected URLs 

- https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing_dod.html

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: guides/v2.1/contributor-guide/contributing_dod.md [hunk 1]
```diff
@@ -166,6 +166,10 @@ The reviewer should check whether the task meets the original acceptance criteri
 * All changes are documented properly
 * All changes are covered with automated tests
 * Determine if code changes caused any failure in continuous integration builds
+* All backward-incompatible changes on code level are approved by an architect. The following cases currently cause SVC failures but do NOT require architect approval:
+  * Adding optional parameter to the constructor, except for the classes explicitly ment for extension, like `AbstractExtensibleModel`
```
Comment (buskamuza) [inline guides/v2.1/contributor-guide/contributing_dod.md]: We need a specific list of classes here.

File: guides/v2.1/contributor-guide/contributing_dod.md [hunk 2]
```diff
@@ -166,6 +166,20 @@ The reviewer should check whether the task meets the original acceptance criteri
 * All changes are documented properly
 * All changes are covered with automated tests
 * Determine if code changes caused any failure in continuous integration builds
+* All backward-incompatible changes on code level are approved by an architect. The following cases currently cause SVC failures but do NOT require architect approval:
```
Comment (buskamuza) [inline guides/v2.1/contributor-guide/contributing_dod.md]: Would be good to move it closer to "There are no unauthorized backward-incompatible" item and maybe phrase those similarly.

Comment (joni-jones) [reply guides/v2.1/contributor-guide/contributing_dod.md]: It's better at first to improve SVC tool and do not trigger failures for such cases.

Comment (paliarush) [reply guides/v2.1/contributor-guide/contributing_dod.md]: @buskamuza done

**Comment by dobooth** [comment]
running tests

File: guides/v2.1/contributor-guide/contributing_dod.md [hunk 3]
```diff
@@ -163,6 +163,20 @@ A task, when complete by the author, should be reviewed by another Magento commu
...
+    * `\Magento\Framework\Model\AbstractExtensibleModel`
+    * `\Magento\Framework\Api\AbstractExtensibleObject`
+    * `\Magento\Framework\Api\AbstractSimpleObject`
+    * `\Magento\Framework\Model\AbstractModel`
+    * `\Magento\Framework\App\Action\Action`
+    * `\Magento\Backend\App\Action`
+    * `\Magento\Backend\App\AbstractAction`
+    * `\Magento\Framework\App\Action\AbstractAction`
+    * `\Magento\Framework\View\Element\AbstractBlock`
+    * `\Magento\Framework\View\Element\Template`
```
Comment (melnikovi) [inline guides/v2.1/contributor-guide/contributing_dod.md]: Would it makes sense to add `\Magento\Framework\Data\Collection\AbstractDb`, `\Magento\Eav\Model\Entity\Collection\AbstractCollection`, `\Magento\Eav\Model\Entity\Collection\VersionControl\AbstractCollection` to the list?

Comment (joni-jones) [reply guides/v2.1/contributor-guide/contributing_dod.md]: Not sure about `\Magento\Eav\Model\Entity\Collection\AbstractCollection` and `\Magento\Eav\Model\Entity\Collection\VersionControl\AbstractCollection` as they are module specific classes.

Comment (melnikovi) [reply guides/v2.1/contributor-guide/contributing_dod.md]: You extend these if your entity has EAV attributes.

**Pull request merged**